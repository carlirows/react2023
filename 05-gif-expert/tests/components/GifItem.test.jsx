import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem.jsx';

describe('GifItem component tests', () => {

    const title = 'A title';
    const url = 'https://localhost/algo.jpg';

    test('should match against snapshot', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();        
    });

    test('sould display image with given "alt" and "url"', () => {

        render( <GifItem title={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        console.log(screen.getByRole('img'));
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('should display title', () => {
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText(title) ).toBeTruthy();
    });
});
