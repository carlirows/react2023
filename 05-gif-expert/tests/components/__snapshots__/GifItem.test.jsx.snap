// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component tests should match against snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A title"
      src="https://localhost/algo.jpg"
    />
    <p>
      A title
    </p>
  </div>
</div>
`;
